### DISCUSSION TITLE
Swap Component for Dark mode

### DISCUSSION THREAD

**Discussion by cvkmohan** [saadeghi/daisyui#808] [Q&A] [answered]
I was wondering if the `swap` component can be used for dark mode flipping. I tried setting the checkbox value to true, but, it is not showing any impact on the mode. Adding and removing the `dark` class is done via Javascript. 
Any guidance would be really helpful.

**Comment by wuarmin** [comment, marked as answer]
It's just the design you are using. Daisy Ui is a CSS Framework. Checkout https://github.com/saadeghi/theme-change

**Reply by cvkmohan** [reply]
My apologies. I could not ask the question properly. I want to use the `theme-change` plugin. My question is
```
        <label class="swap swap-rotate">
          <input type="checkbox" />
          <i class="w-6 h-6 iconify swap-on" data-icon="tabler:sun" data-set-theme="light" />
          <i class="w-6 h-6 iconify swap-off" data-icon="tabler:moon" data-set-theme="dark" />
        </label>
```
I want to use the `swap` component in the above fashion. However, it is not working. Can I make it work?

**Reply by saadeghi** [reply]
@cvkmohan Here's an example of this code snippet, working: https://stackblitz.com/edit/daisyui-cdn-5jgm7n?file=index.html

**Reply by cvkmohan** [reply]
Awesome @saadeghi . Thanks a lot. Such prompt response. 
Just for the record,
I think, my snippet was not working because, of that `data-icon="tabler:sun"` - the `iconify` javascript was probably interfering with event propagation.
After your snippet, I changed the icon to `svg` and it started working.
